### PR TITLE
Run java serially

### PR DESCRIPTION
**Pull request by ineiti**
When I tried to use it, it produced a lot of errors while downloading the google formatter. From what
I can gather, the following happens when a lot of files are formatted:
- pre-commit calls `pretty-format-java` in parallel with a bunch of files in each call
- each `pretty-format-java` tries to download the google formatter
- all but one fail

Most probably the other formatters should also have `require_serial`...

### CODE REVIEW

**Review by cortinico — APPROVED** [review]
Thanks for the contrib @ineiti

**Comment by macisamuele** [comment]
@ineiti Thanks a lot for reporting it.
I've got a pretty busy week to check into it, but I've tried to replicate it and I was able to extract the following traceback
```
Traceback (most recent call last):
  File ".tox/py38/bin/pretty-format-java", line 11, in <module>
    load_entry_point('language-formatters-pre-commit-hooks', 'console_scripts', 'pretty-format-java')()
  File "<decorator-gen-4>", line 2, in pretty_format_java
  File "/Users/maci/github/language-formatters-pre-commit-hooks/language_formatters_pre_commit_hooks/pre_conditions.py", line 31, in java_required
    return f(*args, **kwargs)
  File "/Users/maci/github/language-formatters-pre-commit-hooks/language_formatters_pre_commit_hooks/pretty_format_java.py", line 48, in pretty_format_java
    google_java_formatter_jar = download_google_java_formatter_jar()
  File "/Users/maci/github/language-formatters-pre-commit-hooks/language_formatters_pre_commit_hooks/pretty_format_java.py", line 26, in download_google_java_formatter_jar
    return download_url(get_url(version))
  File "/Users/maci/github/language-formatters-pre-commit-hooks/language_formatters_pre_commit_hooks/utils.py", line 65, in download_url
    os.rename(tmp_file, final_file)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/maci/.cache/pre-commit/google-java-format-1.9-all-deps.jar_tmp' -> '/Users/maci/.cache/pre-commit/google-java-format-1.9-all-deps.jar'
```

Essentially when there are parallel downloads we download on the same file that is cleaned one the download finishes, but this causes the other downloaders to fail on renaming the file.

A possible approach would be to really use temporary files instead of a file suffixed with `_tmp` and/or have a smarter logic to avoid multiple downloads.

I'm more than happy to keep serial execution, after all I don't expect that we're running the hook over a huge amount of files into a single commit all the times, so timing wise should be fine either way.

The TL;DR: of all of this is: the fix is valid and thanks for have uncovered a lurking issue in the tool ;)

**Comment by ineiti** [comment]
> A possible approach would be to really use temporary files instead of a file suffixed with `_tmp` and/or have a smarter logic to avoid multiple downloads.

That will work - but you'd download the file multiple times, at least on the first run... 

Looking at the pre-commit plugin definition, each plugin is installed using `pip install .` - so this looks like a better place to download external dependencies. You're sure this is run only once, and you wouldn't need to check when running the plugin itself.

But, well, never change a running system ;)
